.github/workflows/mobile_job.yml

```diff
@@ -108,10 +108,23 @@ jobs:
           mkdir -p "${GITHUB_WORKSPACE}"
           echo "::endgroup::"
 
+      - name: Calculate the role duration (seconds)
+        id: calculate-role-duration
+        env:
+          TIMEOUT_IN_MINUTES: ${{ inputs.timeout }}
+        shell: bash
+        run: |
+          set -ex
+
+          TIMEOUT_IN_SECONDS=$(( TIMEOUT_IN_MINUTES * 60 ))
```

Comment:
The role duration probably should be somewhat longer than the bare timeout. Say 1h more than the timeout. But I wouldn't go with a complex solution here, instead just asking for the longest duration you can ask. I don't see any downsides of doing this, and it is less prone to failures.

Reply:
Sounds good, let me just stick the max duration here to keep it simple